### PR TITLE
fix(generic-worker): copy archive to task user dir before unarchiving

### PR DESCRIPTION
**Pull request by matt-boris**
>Generic Worker now copies archives to the task user's directory before unarchiving.

### CODE REVIEW

File: workers/generic-worker/mounts.go [hunk 1]
```diff
@@ -724,14 +724,20 @@ func extract(fsContent FSContent, format string, dir string, taskMount *TaskMoun
 	if err != nil {
 		return err
 	}
-	err = makeFileReadWritableForTaskUser(taskMount, cacheFile)
+	copyToPath := filepath.Join(taskContext.TaskDir, filepath.Base(cacheFile))
+	defer os.Remove(copyToPath)
+	_, err = fileutil.Copy(copyToPath, cacheFile)
```
Comment (petemoore) [inline workers/generic-worker/mounts.go:729]: Would be good to log that the worker is copying the file here, e.g.

```go
taskMount.Infof("Copying file '%v' to '%v'", cacheFile, copyToPath)
```

File: workers/generic-worker/mounts.go [hunk 2]
```diff
@@ -724,14 +724,20 @@ func extract(fsContent FSContent, format string, dir string, taskMount *TaskMoun
 	if err != nil {
 		return err
 	}
-	err = makeFileReadWritableForTaskUser(taskMount, cacheFile)
+	copyToPath := filepath.Join(taskContext.TaskDir, filepath.Base(cacheFile))
+	defer os.Remove(copyToPath)
```
[the remark this answers is not in the document]
Comment (matt-boris) [reply workers/generic-worker/mounts.go:728]: Thanks Pete.. I was thinking about doing this, but decided on not doing so for whatever reason :) I'll add it in!

File: workers/generic-worker/mounts.go [hunk 3]
```diff
@@ -714,24 +714,37 @@ func ensureCached(fsContent FSContent, taskMount *TaskMount) (file string, err e
 	return
 }
 
-func extract(fsContent FSContent, format string, dir string, taskMount *TaskMount) error {
+func extract(fsContent FSContent, format string, dir string, taskMount *TaskMount) (err error) {
 	cacheFile, err := ensureCached(fsContent, taskMount)
```
Comment (petemoore) [inline workers/generic-worker/mounts.go:718]: I think this `err` will shadow the one in the function signature, so to be safe I think you will need to change this to:
```go
string cacheFile
cacheFile, err = ensureCached(fsContent, taskMount)
```

Comment (matt-boris) [reply workers/generic-worker/mounts.go:718]: Ahh crap, good catch thank you

File: workers/generic-worker/mounts.go [hunk 4]
```diff
@@ -714,24 +714,38 @@ func ensureCached(fsContent FSContent, taskMount *TaskMount) (file string, err e
...
 }
 
-func extract(fsContent FSContent, format string, dir string, taskMount *TaskMount) error {
-	cacheFile, err := ensureCached(fsContent, taskMount)
+func extract(fsContent FSContent, format string, dir string, taskMount *TaskMount) (err error) {
+	var cacheFile string
+	cacheFile, err = ensureCached(fsContent, taskMount)
 	if err != nil {
 		log.Printf("Could not cache content: %v", err)
 		return err
```
Comment (petemoore) [inline workers/generic-worker/mounts.go:722]: nit: `return` would be sufficient here, but it doesn't really matter

**Review by petemoore — APPROVED** [review]
Thanks Boris, looks awesome!

File: workers/generic-worker/mounts.go (hunk 4 above)
Comment (matt-boris) [reply workers/generic-worker/mounts.go:722]: 😅 I meant to remove these, just missed one! Thank you

**Review by petemoore — APPROVED** [review]

**Comment by petemoore** [comment]
Awesome job!

**Pull request merged**